From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 3 Apr 2020 16:43:01 +0200
Subject: wallpaper: Set the picture style when setting the URI

Use the same picture style as when setting picture wallpapers in the
Background Settings.

Origin: upstream, 1.7.2, commit:7adf5f6b899d740fea2e337e55ff888046e805a7
---
 configure.ac    | 2 +-
 src/wallpaper.c | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 41d103b..9ebb3e5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -57,7 +57,7 @@ AC_SUBST([DBUS_INTERFACES_DIR], [`$PKG_CONFIG --variable=interfaces_dir dbus-1`]
 AC_SUBST([GDBUS_CODEGEN], [`$PKG_CONFIG --variable gdbus_codegen gio-2.0`])
 AC_SUBST([GLIB_COMPILE_RESOURCES], [`$PKG_CONFIG --variable glib_compile_resources gio-2.0`])
 
-PKG_CHECK_MODULES(GTK, [xdg-desktop-portal >= 1.5 glib-2.0 >= 2.44 gio-unix-2.0 gtk+-3.0 >= 3.14 gtk+-unix-print-3.0 fontconfig gnome-desktop-3.0])
+PKG_CHECK_MODULES(GTK, [xdg-desktop-portal >= 1.5 glib-2.0 >= 2.44 gio-unix-2.0 gtk+-3.0 >= 3.14 gtk+-unix-print-3.0 fontconfig gnome-desktop-3.0 gsettings-desktop-schemas])
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
diff --git a/src/wallpaper.c b/src/wallpaper.c
index 62c5534..b2e22b3 100644
--- a/src/wallpaper.c
+++ b/src/wallpaper.c
@@ -14,6 +14,7 @@
 
 #include <glib/gi18n.h>
 #include <gio/gdesktopappinfo.h>
+#include <gdesktop-enums.h>
 
 #include "xdg-desktop-portal-dbus.h"
 #include "shell-dbus.h"
@@ -77,9 +78,8 @@ set_gsettings (gchar *schema,
 
   settings = g_settings_new (schema);
 
-  return g_settings_set_string (settings,
-                                "picture-uri",
-                                uri);
+  return (g_settings_set_string (settings, "picture-uri", uri) &&
+          g_settings_set_enum (settings, "picture-options", G_DESKTOP_BACKGROUND_STYLE_ZOOM));
 }
 
 static void
